src/integration-test/java/com/mongodb/jdbc/integration/testharness/DataLoader.java

```diff
@@ -132,49 +139,10 @@ public void loadTestData() throws IOException {
             try (MongoClient mongoClient = new MongoClient(mdbUri)) {
                 for (TestDataEntry entry : datasets) {
                     MongoDatabase database = mongoClient.getDatabase(entry.db);
-                    MongoCollection<Document> collection = database.getCollection(entry.collection);
-
-                    if (entry.docsExtJson != null) {
-                        // Process extended json format
-                        for (Map<String, Object> row : entry.docsExtJson) {
-                            Document d = Document.parse(new Document(row).toJson());
-                            collection.insertOne(new Document(d));
-                        }
-                        System.out.println(
-                                "Inserted "
-                                        + entry.docsExtJson.size()
-                                        + " rows into "
-                                        + entry.db
-                                        + "."
-                                        + entry.collection);
-                    } else if (entry.docs != null) {
-                        for (Map<String, Object> row : entry.docs) {
-                            collection.insertOne(new Document(row));
-                        }
-                        System.out.println(
-                                "Inserted "
-                                        + entry.docs.size()
-                                        + " rows into "
-                                        + entry.db
-                                        + "."
-                                        + entry.collection);
-                    }
-                    if (entry.nonuniqueIndexes != null) {
-                        for (Map<String, Object> index : entry.nonuniqueIndexes) {
-                            String indexName = collection.createIndex(new Document(index));
-                            System.out.println(
-                                    "Created index "
-                                            + indexName
-                                            + " on "
-                                            + entry.db
-                                            + "."
-                                            + entry.collection);
-                        }
-                    }
-                    if (entry.schema != null) {
-                        setSchema(entry.db, entry.collection, entry.schema);
-                    } else {
-                        generateSchema(entry.db, entry.collection);
+                    if (entry.collection != null) {
+                        loadCollection(entry, database);
+                    } else if (entry.view != null) {
+                        generateSchema(entry.db, entry.view);
```

Comment:
Does the view schema have to be generated after the source collection is loaded and has a generated schema? In our case `bar`.  If so, then we may need to add logic to ensure the view schema is generated after the collections are loaded/schema generated.  
Or if the items in the yaml file is processed in the order it is in the file, it may be enough to add the `baz` view entry at the end of dbmd_variable_result_set_data.yml since that's where the `bar` collection is in.

Reply:
ah, good point. I'll add some more logic to handle views in the end (probably bringing back the separate list).

Reply:
Done!